sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -106,28 +106,34 @@ private void runChecks(SensorContext context, XmlFile xmlFile, NewXmlFile newXml
     checks.all().stream()
       .filter(NewXmlCheck.class::isInstance)
       .map(NewXmlCheck.class::cast)
-      .forEach(check -> runCheck(context, check, newXmlFile));
+      .forEach(check -> {
+        RuleKey ruleKey = checks.ruleKey(check);
+        if (ruleKey == null) {
+          throw new IllegalStateException("Failed to get rule key for rule " + check.getClass());
```

Comment:
If we fail here, we are most probably going to fail on each file everytime we try to execute this rule. Do we want to log something for each file?